BDArmory/Settings/CompSettings.cs

```diff
@@ -84,16 +93,24 @@ public static void Load()
                     }
                 }
             }
+            string VCComment = "Set construction rule limits here. maxStacking is Wing Stacking% in the SPH/VAB BDA Utilities Tool GUI. Use -1 for no limit for the respective field. Setting maxEngines to a negative number will mandate a minimum engine count, e.g. maxEngines = -2 requires 2+ engines on the craft."; // Note: reading the node doesn't seem to get the comment, so we need to reset it each time.
+            if (!fileNode.HasNode("VesselChecks"))
+            {
+                Initialize("VesselChecks", vesselChecks, VCComment, fileNode);
+            }
             if (fileNode.HasNode("VesselChecks"))
             {
-                CompVesselChecksEnabled = true;
                 ConfigNode settings = fileNode.GetNode("VesselChecks");
-
+                settings.comment = VCComment;
                 foreach (ConfigNode.Value fieldNode in settings.values)
                 {
                     if (float.TryParse(fieldNode.value, out float fieldValue))
+                    {
                         vesselChecks[fieldNode.name] = fieldValue; // Add or set the override.
+                        if ((fieldNode.name != "maxEngines" && fieldValue > 0) || fieldValue != 999) CompVesselChecksEnabled = true;
```

Comment:
This condition is always going to enable comp vessel checks. I think you want the following instead
```csharp
switch (fieldNode.name)
{
    case "maxEngines": if (fieldValue != 999) CompVesselChecksEnabled = true; break;
    default: if (fieldValue >= 0) CompVesselChecksEnabled = true; break;
}
```